File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_login_log.ts
import service from "@liujitcn/kratos-admin-core/request";
import type {
  BaseLoginLog,
  BaseLoginLogService,
  GetBaseLoginLogRequest,
  PageBaseLoginLogRequest,
  PageCurrentUserLoginLogRequest,
  PageBaseLoginLogResponse
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_login_log";

const BASE_LOGIN_LOG_URL = "/v1/admin/base/login-log";

/** Admin 登录日志服务。 */
export class BaseLoginLogServiceImpl implements BaseLoginLogService {
  /** 查询当前用户本人的登录记录。 */
  PageCurrentUserLoginLog(request: PageCurrentUserLoginLogRequest): Promise<PageBaseLoginLogResponse> {
    return service<PageCurrentUserLoginLogRequest, PageBaseLoginLogResponse>({
      url: `${BASE_LOGIN_LOG_URL}/current-user`,
      method: "get",
      params: request
    });
  }

  /** 查询登录日志分页列表。 */
  PageBaseLoginLog(request: PageBaseLoginLogRequest): Promise<PageBaseLoginLogResponse> {
    return service<PageBaseLoginLogRequest, PageBaseLoginLogResponse>({
      url: BASE_LOGIN_LOG_URL,
      method: "get",
      params: request
    });
  }

  /** 查询登录日志详情。 */
  GetBaseLoginLog(request: GetBaseLoginLogRequest): Promise<BaseLoginLog> {
    return service<GetBaseLoginLogRequest, BaseLoginLog>({
      url: `${BASE_LOGIN_LOG_URL}/${request.id}`,
      method: "get"
    });
  }
}

/** 默认登录日志服务。 */
export const defBaseLoginLogService = new BaseLoginLogServiceImpl();

File: backend/internal/service/system/admin/v1/base_login_log.go
package admin

import (
	"context"
	"fmt"

	"github.com/go-kratos/kratos/v3/log"
	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"
)

// BaseLoginLogService 提供登录日志的只读管理接口。
type BaseLoginLogService struct {
	adminv1.UnimplementedBaseLoginLogServiceServer // 提供未实现 RPC 的默认返回。
	baseLoginLogCase                               *biz.BaseLoginLogCase
}

// NewBaseLoginLogService 创建登录日志管理服务。
func NewBaseLoginLogService(baseLoginLogCase *biz.BaseLoginLogCase) *BaseLoginLogService {
	return &BaseLoginLogService{baseLoginLogCase: baseLoginLogCase}
}

// PageCurrentUserLoginLog 查询当前用户本人的登录记录。
func (s *BaseLoginLogService) PageCurrentUserLoginLog(ctx context.Context, req *adminv1.PageCurrentUserLoginLogRequest) (*adminv1.PageBaseLoginLogResponse, error) {
	page, err := s.baseLoginLogCase.PageCurrentUserLoginLog(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageCurrentUserLoginLog %v", err))
		return nil, errorsx.WrapInternal(err, "查询本人登录记录失败")
	}
	return page, nil
}

// PageBaseLoginLog 查询登录日志分页列表。
func (s *BaseLoginLogService) PageBaseLoginLog(ctx context.Context, req *adminv1.PageBaseLoginLogRequest) (*adminv1.PageBaseLoginLogResponse, error) {
	page, err := s.baseLoginLogCase.PageBaseLoginLog(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBaseLoginLog %v", err))
		return nil, errorsx.WrapInternal(err, "查询登录日志分页列表失败")
	}
	return page, nil
}

// GetBaseLoginLog 查询登录日志详情。
func (s *BaseLoginLogService) GetBaseLoginLog(ctx context.Context, req *adminv1.GetBaseLoginLogRequest) (*adminv1.BaseLoginLog, error) {
	item, err := s.baseLoginLogCase.GetBaseLoginLog(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("GetBaseLoginLog %v", err))
		return nil, errorsx.WrapInternal(err, "查询登录日志详情失败")
	}
	return item, nil
}
